modules/kubeflow-feast/tests/conftest.py

```diff
@@ -34,6 +59,21 @@ def pytest_addoption(parser):
         help="Risk to be used when deploying the terraform module",
     )
 
+@pytest.fixture(scope="module")
+def istio_cni_bin_dir(request) -> str:
```

Comment:
```suggestion
def istio_cni_bin_dir(request) -> str | None:
```

I suppose this can be none if the option is not declared. Similarly also below

Reply:
Wouldn't it be better to keep them as strings and only strings, as [the underlying charm interprets only strings and wants empty ones to disable Istio CNI](https://github.com/canonical/istio-operators/blob/main/charms/istio-pilot/config.yaml#L2-L15)? It may be less risk prone to avoid the conversion:
_<ins>only</ins> strings CLI args -> <ins>either</ins> strings <ins>or</ins> Nones in Python -> <ins>only</ins> strings in Terraform (-> <ins>only</ins> strings as charm configs)_

This way we always propagate them as is from the module to the charm instead of risking not to because of Python logic bases on `None`s.

Reply:
As it is written, I believe the type *could* also be None. To make it type compatible just use:

```
return request.config.getoption("--istio-cni-bin-dir", "")
```

and you could also use skip=False to raise

See [here](https://github.com/pytest-dev/pytest/blob/8fb7815f1440c7cf8718c248639109a779db575d/src/_pytest/config/__init__.py#L1877)

Reply:
Agreed with call: `request.config.getoption("--istio-cni-bin-dir") or ""`